src/Material/Footer.elm

```diff
@@ -0,0 +1,590 @@
+> footer forms have their own internal structures, including required and optional
+> elements, and typically include both informational and clickable content, such
+> as links.
+>
+> Footers, as represented by this component, are a fairly new feature in user
+> interfaces, and allow users to view discrete blocks of content in a coherent and
+> consistently organized way. Their design and use is an important factor in the
+> overall user experience.
+
+See also the
+[Material Design Specification](https://material.google.com/layout/structure.html).
+
+Refer to [this site](http://debois.github.io/elm-mdl#/footer)
+for a live demo.
+
+# Types
+
+@docs Type
+@docs Content, Footer, Element
+@docs Property
+
+@docs MegaFooter, MiniFooter
+
+@docs Section
+@docs TopSection, MiddleSection, BottomSection
+@docs Dropdown
+
+# Helpers
+
+@docs wrap
+@docs link, onClick, href
+
+# Appearance
+
+@docs mini, mega
+
+# Sections
+
+@docs left, right, top, bottom, middle
+
+# Content
+
+@docs links, logo, text, socialButton, dropdown, heading, linkItem
+
+-}
+
+import Html exposing (..)
+import Html.Attributes as Html
+import Html.Events as Events
+import Material.Options as Options exposing (Style, cs)
+import String
+import Regex
+import Material.Options.Internal as Internal exposing (attribute)
+
+
+{-| The type of the footer
+-}
+type Type
+  = Mini
+  | Mega
+
+
+prefix : Type -> String
+prefix tp =
+  case tp of
+    Mini ->
+      "mdl-mini-footer"
+
+    Mega ->
+      "mdl-mega-footer"
+
+
+separator : String
+separator =
+  "__"
+
+
+{-| `FooterProperty`
+-}
+type FooterProperty
+  = FooterProperty
+
+
+{-| Property for footers
+-}
+type alias Property m =
+  Options.Property FooterProperty m
+
+
+{-| Footer content can either be Html or footer specific content
+-}
+type Content a
+  = HtmlContent (Html a)
+  | Content (Footer a)
+
+
+{-| Internal type alias for content within a footer
+-}
+type alias Footer a =
+  { styles : List (Property a)
+  , content : List (Content a)
+  , elem : Element a
+  }
+
+
+{-| Helpers alias to wrap a Html element function
+-}
+type alias Element a =
+  List (Html.Attribute a) -> List (Html.Html a) -> Html.Html a
+
+
+{-| Strongly typed `Section` in a footer
+-}
+type Section a
+  = Section (Content a)
+
+
+{-| Strongly typed `Dropdown` in a footer
+-}
+type Dropdown a
+  = Dropdown
+      { props : List (Property a)
+      , content : List (Content a)
+      }
+
+
+{-| Strongly typed `TopSection` in a footer
+-}
+type TopSection a
+  = TopSection
+      { left : Maybe (Section a)
+      , right : Maybe (Section a)
+      , props : List (Property a)
+      }
+
+
+{-| Strongly typed `BottomSection` in a footer
+-}
+type BottomSection a
+  = BottomSection
+      { props : List (Property a)
+      , content : List (Content a)
+      }
+
+
+{-| Strongly typed `MiddleSection` in a footer
+-}
+type MiddleSection a
+  = MiddleSection
+      { props : List (Property a)
+      , content : List (Dropdown a)
```

Comment:
I think it might not actually be true, I think I was basing it on the examples where they placed the dropdowns in the middle, which I think is the preferred place for them, but perhaps not something we need to specifically enforce in this case.